pkg/db/common.go

```diff
@@ -0,0 +1,454 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkgdb
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/mysql"
+)
+
+const (
+	// ImplicitColName is tidb's implicit column's name
+	ImplicitColName = "_tidb_rowid"
+
+	// ImplicitColID is tidb's implicit column's id
+	ImplicitColID = -1
+)
+
+// DBConfig is the DB configuration.
+type DBConfig struct {
+	Host string `toml:"host" json:"host"`
+
+	Port int `toml:"port" json:"port"`
+
+	User string `toml:"user" json:"user"`
+
+	Password string `toml:"password" json:"password"`
+
+	Schema string `toml:"schema" json:"schema"`
+}
+
+// String returns string of database config
+func (c *DBConfig) String() string {
+	if c == nil {
+		return "<nil>"
+	}
+	return fmt.Sprintf("DBConfig(%+v)", *c)
+}
+
+// CreateDB create a mysql fd
+func CreateDB(cfg DBConfig, snapshot string) (*sql.DB, error) {
+	dbDSN := fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?charset=utf8", cfg.User, cfg.Password, cfg.Host, cfg.Port, cfg.Schema)
+	dbConn, err := sql.Open("mysql", dbDSN)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	if snapshot != "" {
+		err = SetSnapshot(dbConn, snapshot)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+
+	return dbConn, nil
+}
+
+// CloseDB closes the mysql fd
+func CloseDB(db *sql.DB) error {
+	return errors.Trace(db.Close())
+}
+
+// GetCreateTableSQL gets the create table sql.
```

Comment:
We can put these functions in package later separately @WangXiangUSTC

Reply:
ok